tests/framework/utils.py

```diff
@@ -56,61 +55,52 @@ def set_cpu_affinity(pid: int, cpulist: list) -> list:
     return psutil.Process(pid).cpu_affinity(real_cpulist)
 
 
-def get_thread_name(pid: int, tid: int) -> str:
-    """Return thread name from pid and tid pair."""
-    try:
-        return (
-            Path("/proc", str(pid), "task", str(tid), "comm").read_text("utf-8").strip()
-        )
-    except FileNotFoundError as exc:
-        raise psutil.NoSuchProcess(tid) from exc
-
-
 CpuTimes = namedtuple("CpuTimes", ["user", "system"])
 
 
-def get_cpu_times(pid: int) -> Dict[str, CpuTimes]:
-    """Return a dict mapping thread name to CPU usage (in seconds) since start."""
-    threads = []
-    try:
-        threads = psutil.Process(pid).threads()
-    except psutil.NoSuchProcess as exc:
-        logging.warning("Process %d does not exist", pid, exc_info=exc)
-        return {}
-
-    cpu_times = {}
-    for thread in threads:
-        try:
-            thread_name = get_thread_name(pid, thread.id)
-            cpu_times[thread_name] = CpuTimes(thread.user_time, thread.system_time)
-        except psutil.NoSuchProcess as exc:
-            logging.warning("Thread %d no longer exists", thread.id, exc_info=exc)
-            continue
-
-    return cpu_times
-
-
-def get_cpu_utilization(
-    pid: int,
-    interval: int = 1,
-    split_user_system: bool = False,
-) -> Dict[str, float | CpuTimes]:
-    """Return current process per thread CPU utilization over the interval (seconds)."""
-    cpu_utilization = {}
-    cpu_times_before = get_cpu_times(pid)
-    time.sleep(interval)
-    cpu_times_after = get_cpu_times(pid)
-    threads = set(cpu_times_before.keys()) & set(cpu_times_after.keys())
-    for thread_name in threads:
-        before = cpu_times_before[thread_name]
-        after = cpu_times_after[thread_name]
-        user = (after.user - before.user) / interval * 100
-        system = (after.system - before.system) / interval * 100
-        if split_user_system:
-            cpu_utilization[thread_name] = CpuTimes(user, system)
-        else:
-            cpu_utilization[thread_name] = user + system
-    return cpu_utilization
+class ProcessExt(psutil.Process):
+    """Extends psutil.Process with better thread support."""
+
+    def threads_cpu_times(self) -> Dict[str, CpuTimes]:
+        """Return a dict mapping thread name to CPU usage (in seconds) since start."""
+        cpu_times = {}
+        for pthread in super().threads():
```

Comment:
why do we need `super()` here? we're not overwriting any methods :thinking: 

```suggestion
        for pthread in self.threads():
```

Reply:
yeah I had a few reworks here and was overriding the `threads()` initially. I'll change this if I have to push another revision

Reply:
I've removed the class so it's no longer relevant